train_spaceenv.py

```diff
@@ -240,48 +240,65 @@ def main(cfg: DictConfig):
     treat_featimp = (treat_featimp / tscale).importance.to_dict()
     treat_featimp = {c: float(treat_featimp.get(c, 0.0)) for c in covariates}
 
-    # === Compute confounding score ===
-    scoring_method = cfg.confounding_score_method
-    logging.info(f"Computing confounding score using {scoring_method}.")
-
-    if scoring_method == "feat_importance":
-        # compute the confounding score
-        confounding_score = {k: min(treat_featimp[k], featimp[k]) for k in covariates}
-        logging.info(f"Combined importances with min:\n{confounding_score}")
-
-    elif scoring_method == "leave_out_fit":
-        confounding_score = {}
-
-        for i, g in enumerate(covar_groups):
-            key_ = list(g.keys())[0] if isinstance(g, dict) else g
-            value_ = list(g.values())[0] if isinstance(g, dict) else [g]
-            cols = dftrain.columns.difference(value_)
-            leave_out_predictor = TabularPredictor(label=spaceenv.outcome)
-            leave_out_predictor = leave_out_predictor.fit(
-                train_data[cols],
-                **spaceenv.autogluon.fit,
-                tuning_data=tuning_data[cols],
-                use_bag_holdout=True,
-            )
-            leave_out_predictor.refit_full()
-
-            leave_out_mu_cf = []
-            for a in avals:
-                cfdata = df[cols].copy()
-                cfdata[spaceenv.treatment] = a
-                predicted = leave_out_predictor.predict(TabularDataset(cfdata))
-                leave_out_mu_cf.append(predicted)
-            leave_out_mu_cf = pd.concat(leave_out_mu_cf, axis=1)
-
-            # compute loss normalized by the variance of the outcome
-            rss = np.mean((leave_out_mu_cf.values - mu_cf.values) ** 2)
-            # score = rss / mu_cf.values.var()
-            score = rss / yscale
-            confounding_score[key_] = score
-            logging.info(f"[{i + 1} / {len(covar_groups)}] {key_}: {score:.4f}")
+    # legacy confounding score by inimum
+    cs_minimum = {k: min(treat_featimp[k], featimp[k]) for k in covariates}
+    logging.info(f"Legacy conf. score by minimum:\n{cs_minimum}")
+
+    # === Compute confounding scores ===
+    # The strategy for confounding scores is to compute various types
+    # using the baseline model.
+
+    # For continous treatment compute the ERF and PEHE scores
+    # For categorical treatmetn additionally compute the ATE score
+    # For both also use the minimum of the treatment and outcome model
+    # As in the first version of the paper.
+
+    # For comparability across environments, we divide the scores by the
+    # variance of the synthetic outcome.
+
+    # Obtain counterfactuals for the others
+    cs_erf = {}
+    cs_pehe = {}
+    cs_ate = {}  # will be empty if not binary
+    scale = np.var(Y_synth)
```

Comment:
I think (np.mean(Y_synth))^ 2 makes more sense because some variables could have high variance and this variance won't mean the causal effect is large or small.

Reply:
Upon second thoughts, using the mean wouldn't make much sense. Here's an example with potential outcomes:

potential outcomes scenario 1:
$Y^a_i = \tau \cdot a + \beta \cdot  X + \epsilon_i $

scenario 2, same but shifted up:
$Y^a_i =  \tau \cdot a  + \beta \cdot X + \epsilon_i + \alpha$.

And assume $E[X]=E[\epsilon]=0$.
In both cases, the ATE is $\tau$. In scenario 1, $E[Y]=0$ and in scenario 2 $E[Y]=\alpha$. 

Observations:
1. If we normalize by $E[Y]^2$ as suggested, then the normalized ATE in scenario 1 would be $tau/0=\infty$. 
2. Scenarios 1 and 2 are just a constant shift. The magnitude of the ATE is the same, and we would need a normalization such that they are comparable.  Normalizing by $\sigma^2=\sqrt{Var(Y)}$ would give the same normalized ATE in both scenarios.